data/mods/gen7/random-teams.ts

```diff
@@ -200,15 +201,23 @@ export class RandomGen7Teams extends RandomTeams {
 				case 'healingwish': case 'memento':
 					if (counter.setupType || !!counter['recovery'] || hasMove['substitute']) rejected = true;
 					break;
+				case 'helpinghand':
+					if (counter.setupType) rejected = true;
+					break;
+				case 'icywind': case 'stringshot':
+					if (!!counter['speedsetup'] || hasMove['trickroom']) rejected = true;
+					break;
 				case 'leechseed': case 'roar': case 'whirlwind':
 					if (counter.setupType || !!counter['speedsetup'] || hasMove['dragontail']) rejected = true;
+					if (isDoubles && movePool.includes('protect') || movePool.includes('spikyshield')) rejected = true;
 					break;
 				case 'nightshade': case 'seismictoss': case 'superfang':
 					if (!hasAbility['Parental Bond'] && (counter.damagingMoves.length > 1 || counter.setupType)) rejected = true;
 					break;
 				case 'protect':
 					if (counter.setupType && !hasMove['wish']) rejected = true;
 					if (hasMove['rest'] || hasMove['lightscreen'] && hasMove['reflect']) rejected = true;
+					if (isDoubles && (hasMove['tailwind'] && hasMove['roost']) || hasMove['fakeout'] || movePool.includes('bellydrum') || movePool.includes('shellsmash')) rejected = true;
```

Comment:
What? If you're rejecting Protect basically all the time then just remove it from the move pools...

Reply:
we still want protect; we just don't want mons to not have Belly Drum/Shell Smash when they otherwise could, and we want to avoid Roost + Tailwind + Protect specifically